WalletWasabi.Fluent/ViewModels/Wallets/Transactions/Inputs/InputsCoinViewModel.cs

```diff
@@ -0,0 +1,53 @@
+using System.Linq;
+using Avalonia;
+using Avalonia.Controls;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Fluent.Models.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Transactions.Inputs;
+
+public class InputsCoinViewModel : InputsCoinListItem
+{
+    public InputsCoinViewModel(SmartCoin coin)
+	{
+		Coin = coin;
+		Amount = new Amount(coin.Amount);
+		if(coin.HdPubKey.HistoricalAnonSet.TryGetValue(coin.Outpoint.Hash, out var anonSetWhenTxProcessed))
+		{
+			AnonymityScore = (int)anonSetWhenTxProcessed;
+		}
+		else
+		{
+			AnonymityScore = (int)coin.AnonymitySet;
+		}
+	}
+
+	public InputsCoinViewModel(InputsCoinViewModel[] coins, int inputCount, bool isExpanded, int? nbDiff)
+	{
+		Amount = new Amount(coins.Sum(x => x.Amount.Btc));
+		Children = coins;
+		TotalInputs = inputCount;
+		IsExpanded = isExpanded;
+		NbDiff = nbDiff;
+		if (IsExpanded)
+		{
+			foreach (var coin in Children)
+			{
+				coin.IsExpanded = true;
+			}
+		}
+
+		if (Children.Count == TotalInputs)
+		{
+			TitleText = $"{Children.Count} input{(Children.Count == 1 ? "" : "s")}";
+			Margin = new Thickness(45, 0, 0, 0);
+		}
+		else
+		{
+			TitleText = $"{Children.Count} own out of {TotalInputs} input{(TotalInputs == 1 ? "" : "s")}";
```

Comment:
this is a bit odd in the UI imo: "`0 own out of 21 inputs`"

Maybe could remove this. Just have `X inputs` and leave the tooltip, which then explains why it cannot be expanded

Reply:
I followed your advice